tt_metal/hostdevcommon/debug_ring_buffer_common.h

```diff
@@ -0,0 +1,16 @@
+// SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+// We use a magic value to initialize the rung buffer to, so that we can avoid printing it to the
+// watcher og if no ring buffer data has been written.
+constexpr int DEBUG_RING_BUFFER_STARTING_MAGIC = 0x12341234;
+
+constexpr static std::uint32_t RING_BUFFER_ELEMENTS = RING_BUFFER_SIZE / sizeof(uint32_t) - 1;
+struct DebugRingBufMemLayout {
+    uint32_t current_ptr;
```

Comment:
can you set current_ptr to -1 rather than using the magic value?  then writing is *(++current_ptr) = xyz w/ a wrap check before